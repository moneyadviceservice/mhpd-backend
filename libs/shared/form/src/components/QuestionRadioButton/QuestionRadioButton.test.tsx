import React from 'react';
import { render, screen } from '@testing-library/react';
import { QuestionRadioButton } from '.';

describe('Question Radio Button component', () => {
  it('renders correctly', () => {
    render(
      <QuestionRadioButton
        name={'test'}
        options={[
          { text: 'Option 1', value: 'option1' },
          { text: 'Option 2', value: 'option2' },
        ]}
      >
        Test
      </QuestionRadioButton>,
    );
    const question = screen.getByTestId('question-radio');
    expect(question).toMatchSnapshot();
  });
});
