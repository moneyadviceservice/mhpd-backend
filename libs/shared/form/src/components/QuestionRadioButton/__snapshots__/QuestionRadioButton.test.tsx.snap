// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Radio Button component renders correctly 1`] = `
<div
  class=""
  data-testid="question-radio"
>
  <legend
    class="mb-2 text-lg text-gray-800 font-bold"
  >
    Test
  </legend>
  <div
    class="mt-3"
  >
    <div
      class="flex flex-wrap relative"
      data-testid="radio-button"
    >
      <input
        class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer"
        id="test-0"
        name="test"
        required=""
        type="radio"
        value="option1"
      />
      <label
        class="align-self-center max-w-[calc(100%-40px)] p-[7px] pl-[12px] cursor-pointer touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full before:border before:border-gray-400 after:content-[''] after:absolute after:top-[8px] after:left-[8px] after:w-0 after:h-0 after:border-[12px] after:border-solid after:border-pink-600 after:rounded-full after:opacity-0 hover:after:opacity-100 hover:after:border-pink-800 active:after:opacity-100 active:after:border-pink-400 peer-focus:before:border-4 peer-focus:before:border-purple-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_4px] peer-focus:before:shadow-yellow-200 peer-checked:after:opacity-100"
        for="test-0"
      >
        Option 1
      </label>
    </div>
  </div>
  <div
    class="mt-3"
  >
    <div
      class="flex flex-wrap relative"
      data-testid="radio-button"
    >
      <input
        class="w-[40px] h-[40px] m-0 opacity-0 cursor-pointer peer"
        id="test-1"
        name="test"
        required=""
        type="radio"
        value="option2"
      />
      <label
        class="align-self-center max-w-[calc(100%-40px)] p-[7px] pl-[12px] cursor-pointer touch-action-manipulation before:content-[''] before:box-border before:absolute before:top-0 before:left-0 before:w-[40px] before:h-[40px] before:bg-transparent before:rounded-full before:border before:border-gray-400 after:content-[''] after:absolute after:top-[8px] after:left-[8px] after:w-0 after:h-0 after:border-[12px] after:border-solid after:border-pink-600 after:rounded-full after:opacity-0 hover:after:opacity-100 hover:after:border-pink-800 active:after:opacity-100 active:after:border-pink-400 peer-focus:before:border-4 peer-focus:before:border-purple-700 peer-focus:before:outline-[3px] peer-focus:before:outline-black peer-focus:before:outline-offset-1 peer-focus:before:shadow-[0_0_0_4px] peer-focus:before:shadow-yellow-200 peer-checked:after:opacity-100"
        for="test-1"
      >
        Option 2
      </label>
    </div>
  </div>
</div>
`;
